test/RunnerUtil.java

```diff
@@ -16,123 +16,97 @@
  *
  */
 
-package com.vaticle.typedb.common.test.server;
+package com.vaticle.typedb.common.test;
 
-import com.vaticle.typedb.common.test.Runner;
-import org.zeroturnaround.exec.ProcessResult;
-import org.zeroturnaround.exec.StartedProcess;
+import org.zeroturnaround.exec.ProcessExecutor;
 
 import java.io.File;
 import java.io.IOException;
+import java.net.InetSocketAddress;
 import java.net.ServerSocket;
 import java.net.Socket;
+import java.nio.file.Files;
 import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
 import java.util.Timer;
 import java.util.TimerTask;
 import java.util.concurrent.CountDownLatch;
-import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
 
-public abstract class TypeDBRunner extends Runner {
+import static org.junit.Assert.fail;
 
-    private static final int SERVER_STARTUP_TIMEOUT_MILLIS = 30000;
-    private static final int SERVER_ALIVE_POLL_INTERVAL_MILLIS = 500;
+public class RunnerUtil {
+
+    private static final String TAR_GZ = ".tar.gz";
+    private static final String ZIP = ".zip";
     private static final int PORT_ALLOCATION_MAX_RETRIES = 15;
+    public static final int SERVER_STARTUP_TIMEOUT_MILLIS = 30000;
+    private static final int SERVER_ALIVE_POLL_INTERVAL_MILLIS = 500;
     private static final int SERVER_ALIVE_POLL_MAX_RETRIES = SERVER_STARTUP_TIMEOUT_MILLIS / SERVER_ALIVE_POLL_INTERVAL_MILLIS;
 
-    protected final Path dataDir;
-    protected final Path logsDir;
-    private StartedProcess serverProcess;
-
-    public TypeDBRunner() throws InterruptedException, TimeoutException, IOException {
-        super();
-        this.dataDir = rootPath.resolve("server").resolve("data");
-        this.logsDir = rootPath.resolve("server").resolve("logs");
+    public static Path unarchive() throws IOException, InterruptedException, TimeoutException {
```

Comment:
it feels a bit strange to read this static function that lives in a `Util` that has so much context about where to find archives/how to unarchive/figure out where to put the unarchived files... seems like a high-level component to me. Maybe we can make it stateful or pass in more arguments to make it more of a "util"?

Reply:
Addressed in f1a1f266d1a6756fb90267dcfa3da483ed894ce6.